### ISSUE TITLE
Streaming output repeats indefinitely (stream=True) when generating responses using openai responses api

### ISSUE THREAD

**Issue by MeetSolanki530** [lmstudio-ai/lmstudio-bug-tracker#1194]
When running a streaming response in LM Studio with `stream=True`, the model produces repetitive output. Instead of continuing the generation logically, it repeats previous paragraphs or sentences indefinitely.

### **Steps to Reproduce**

1. Launch LM Studio 0.3.31 (build 7)

2. Load model: `qwen3-vl-2b-instruct` 

3. Run:
   ```python
   resp = client.responses.create(
       model="qwen3-vl-2b-instruct",
       input="Write a very long novel about otters in space.",
       stream=True
   )

   for event in resp:
       if event.type == "response.output_text.delta":
           print(event.delta, end="", flush=True)
   ```

4. Observe that the model starts repeating earlier paragraphs indefinitely.

### **Expected Behavior**

The model should generate coherent, continuous output, not duplicate previous sections.

### **Actual Behavior**

Streaming output loops endlessly, repeating the same paragraph multiple times.

### **Environment**

* **LM Studio Version:** 0.3.31(Build 7)
* **OS:** Windows 11
* **Model:** `qwen3-vl-2b-instruct`
* **API:** Local inference (`client.responses.create`)

### **Log Entry**

Response I am getting :-

"""## The River of Ice and the Stars

The silence wasn't empty; it was thick, a velvet shroud that pressed against the bones of the *Aurora*, the first vessel to venture beyond the known stars. It wasn't the absence of sound, but the profound, resonant hum of the universe itself – a low thrumming beneath the hulls, a constant, vibrating presence in the void. The *Aurora* was not just a ship; it was a living organism, a vessel of life and light, navigating through this vast, indifferent expanse.

The crew, a diverse band of humans and alien beings, were not merely explorers. They were custodians of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention. The *Aurora* was built to be a sanctuary for life in space, a vessel designed to protect the planet's unique biosphere from the harshness of interstellar travel.

The crew’s mission was simple: explore the outer reaches of the galaxy, searching for signs of life beyond our own solar system. They were not just scientists or engineers; they were stewards of a fragile ecosystem – the delicate balance between the natural world and human intervention.
"""

<img width="1919" height="1029" alt="Image" src="https://github.com/user-attachments/assets/953a5919-22bb-435f-9c0f-f2b7c855ea94" />

**Log Entry** :- 

2025-11-05 22:13:15 [DEBUG]
 Received request: POST to /v1/responses with body  {
  "input": "Write a very long novel about otters in space.",
  "model": "qwen3-vl-2b-instruct",
  "stream": true
}
2025-11-05 22:13:15  [INFO]
 [qwen3-vl-2b-instruct] Streaming response...
2025-11-05 22:13:15 [DEBUG]
 Sampling params:	repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
	top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.100
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
2025-11-05 22:13:15 [DEBUG]
 Sampling: 
logits -> logit-bias -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
Generate: n_ctx = 4096, n_batch = 512, n_predict = -1, n_keep = 19
Cache reuse summary: 5/19 of prompt (26.3158%), 5 prefix, 0 non-prefix
Total prompt tokens: 19
Prompt tokens to decode: 14
BeginProcessingPrompt
2025-11-05 22:13:15  [INFO]
 [qwen3-vl-2b-instruct] Prompt processing progress: 0.0%
2025-11-05 22:13:15  [INFO]
 [qwen3-vl-2b-instruct] Prompt processing progress: 100.0%
2025-11-05 22:13:15 [DEBUG]
 FinishedProcessingPrompt. Progress: 100
2025-11-05 22:13:37  [INFO]
 [LM STUDIO SERVER] Client disconnected. Stopping generation... (If the model is busy processing the prompt, it will finish first.)
2025-11-05 22:13:37 [DEBUG]
 Target model llama_perf stats:
llama_perf_sampler_print:    sampling time =    2011.33 ms /  1913 runs   (    1.05 ms per token,   951.11 tokens per second)
llama_perf_context_print:        load time =    2054.22 ms
llama_perf_context_print: prompt eval time =     169.72 ms /    14 tokens (   12.12 ms per token,    82.49 tokens per second)
llama_perf_context_print:        eval time =   19122.08 ms /  1893 runs   (   10.10 ms per token,    99.00 tokens per second)
llama_perf_context_print:       total time =   21884.58 ms /  1907 tokens
llama_perf_context_print:    graphs reused =          0
llama_memory_breakdown_print: | memory breakdown [MiB]             | total   free    self   model   context   compute    unaccounted |
llama_memory_breakdown_print: |   - CUDA0 (RTX 3050 Ti Laptop GPU) |  4095 =  643 + (1799 =  1050 +     448 +     300) +        1652 |
llama_memory_breakdown_print: |   - Host                           |                  255 =   243 +       0 +      12                |

**Comment by yagil** [comment]
cc @mattjcly @rugvedS07

**Comment by mattjcly** [comment]
Hi @MeetSolanki530, thanks for the bug report. To help us debug further, would you be able to send us a screenshot of the entry for this model in your models page? This will tell us the source and quantization of the model to try and reproduce the issue.

For example:
<img width="1278" height="150" alt="Image" src="https://github.com/user-attachments/assets/611ad4da-9b68-4571-a944-d57039747ae5" />

**Comment by PyThrrrone** [comment]
Same issue here.

**Comment by Cosby5** [comment]
I'm having a similar issue with the "nvidia/nemotron-3-nano" model version "Q3_K_L", it bugs out in a few other models, but this one is the most aggressive one. 

I give it a simple prompt, say "hello" (yes, I talk to it like a regular person, come at me), then ask it my question after it responds, then it will likely respond with a duplicated response (2 or 3 times) within the first few exchanges. If let alone it will begin duplicated more, like it is reading its context and duplicating that (2, then 5, then 8 or more before I stop it from just spewing tokens).

I can edit the prompt and delete the excess, and it will return within a few more prompts. The prompts are logic and coherent, they are just duplicated. On other models I think it's some artifacting because the output loses consistency as the bug persist. 

P.S.
I can open, up a separate issue if you want. Except, I'm in your frontend and haven't figure out where to find the logging (haven't looked for it).

**Comment by tuner** [comment]
I ran into this too. `stream=True` on the Responses API is duplicating structured output for me.

v0.4.9+1, "qwen/qwen3-4b-2507"
